### PR TITLE
 Add cache_fill_bytes, cache_lookup and protocol field extraction from httpRequest.

### PR DESCRIPTION
**Pull request by qingling128**
Fixes https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/issues/271

### CODE REVIEW

File: test/plugin/test_out_google_cloud.rb [hunk 1]
```diff
@@ -418,11 +418,11 @@ def convert_subfields_to_strings(full_hash, fields_to_convert)
     ])
   end
 
-  # 'responseSize' and 'requestSize' are Integers in the gRPC protos, yet
-  # Strings in REST API client libraries.
+  # 'responseSize', 'requestSize' and 'cacheFillBytes' are Integers in the gRPC
```
Comment (igorpeshansky) [inline test/plugin/test_out_google_cloud.rb]: Oxford comma, please.

File: test/plugin/test_out_google_cloud.rb [hunk 2]
```diff
@@ -418,11 +418,11 @@ def convert_subfields_to_strings(full_hash, fields_to_convert)
...
+  # protos, yet Strings in REST API client libraries.
   def http_request_message
     convert_subfields_to_strings(
-      HTTP_REQUEST_MESSAGE, %w(responseSize requestSize))
+      HTTP_REQUEST_MESSAGE, %w(responseSize requestSize cacheFillBytes))
```
Comment (igorpeshansky) [inline test/plugin/test_out_google_cloud.rb]: Do you want to keep these alphabetical as well?

Comment (qingling128) [reply test/plugin/test_out_google_cloud.rb]: Done

File: test/plugin/test_out_google_cloud.rb (hunk 1 above)
Comment (qingling128) [reply test/plugin/test_out_google_cloud.rb]: Done

**Pull request merged**